From ee855b15ff185bad3b0495500ab234b9cab7436c Mon Sep 17 00:00:00 2001
From: Martin Whitaker <fsf@martin-whitaker.me.uk>
Date: Mon, 24 Feb 2020 22:11:29 +0000
Subject: [PATCH] grub-install: fix inverted test for NLS enabled when copying
 locales.

Commit 3d8439da8c9a4acf9bc1c41b364ec6e1680ef052 attempted to avoid
copying locale files to the target directory when NLS was disabled.
However the test is inverted, and it does the opposite.

Signed-off-by: Martin Whitaker <fsf@martin-whitaker.me.uk>

diff --git a/util/grub-install-common.c b/util/grub-install-common.c
index ca0ac612a..f53bf0694 100644
--- a/util/grub-install-common.c
+++ b/util/grub-install-common.c
@@ -598,7 +598,7 @@ copy_all (const char *srcd,
   grub_util_fd_closedir (d);
 }
 
-#if !(defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
+#if (defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
 static const char *
 get_localedir (void)
 {
@@ -659,7 +659,7 @@ static void
 grub_install_copy_nls(const char *src __attribute__ ((unused)),
                       const char *dst __attribute__ ((unused)))
 {
-#if !(defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
+#if (defined (GRUB_UTIL) && defined(ENABLE_NLS) && ENABLE_NLS)
   char *dst_locale;
 
   dst_locale = grub_util_path_concat (2, dst, "locale");
-- 
2.21.1
